frontend/src/bundles/notifications/notifications-api.ts

```diff
@@ -0,0 +1,94 @@
+import { type Http } from '~/framework/http/types/types.js';
+import { BaseHttpApi } from '~/framework/http-api/http-api.js';
+import { type Storage } from '~/framework/storage/storage.js';
+
+import { NotificationsApiPath } from './enums/enums.js';
+import {
+    type NotificationRequestDto,
+    type NotificationResponseDto,
+} from './types/types.js';
+
+type Constructor = {
+    baseUrl: string;
+    http: Http;
+    storage: Storage;
+};
+
+class NotificationsApi extends BaseHttpApi {
+    public constructor({ baseUrl, http, storage }: Constructor) {
+        super({ path: NotificationsApiPath.ROOT, baseUrl, http, storage });
+    }
+
+    public async fetchNotifications(): Promise<Array<NotificationResponseDto>> {
+        const response = await this.load(
+            this.getFullEndpoint(NotificationsApiPath.ROOT, {}),
+            {
+                method: 'GET',
+                contentType: 'application/json',
+                hasAuth: true,
+            },
+        );
+
+        return await response.json();
+    }
+
+    public async createNotification(
+        payload: NotificationRequestDto,
+    ): Promise<NotificationResponseDto> {
+        const response = await this.load(
+            this.getFullEndpoint(NotificationsApiPath.ROOT, {}),
+            {
+                method: 'POST',
+                contentType: 'application/json',
+                hasAuth: true,
+                payload: JSON.stringify(payload),
+            },
+        );
+
+        return await response.json();
+    }
+
+    public async dismissNotification(notificationId: string): Promise<void> {
+        await this.load(
+            this.getFullEndpoint(
+                `${NotificationsApiPath.DISMISS.replace(':notificationId', notificationId)}`,
+                {},
```

Comment:
```suggestion
            this.getFullEndpoint(NotificationsApiPath.DISMISS, { notificationId }),
```
Also I would suggest to work with id as number everywhere, and only here convert it to string